### PR TITLE
zib-Vaccination

### PR DESCRIPTION
**Pull request by ArdonToonstra**

### CODE REVIEW

File: resources/zib/zib-Vaccination-event.xml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  <type value="Immunization" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Immunization" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Immunization">
+      <path value="Immunization" />
+      <short value="Vaccination" />
+      <comment value="The zib Vaccination is mapped to this Immunization profile and a profile on ImmunizationRecommendation (http://nictiz.nl/fhir/StructureDefinition/zib-Vaccination-request) to align with the intention of FHIR. All past vaccinations are covered using this Immunization profile, while all future vaccinations, including the advised vaccinations, are covered in the ImmunizationRecommendation profile. Both profiles contain the zib mappings, with the exception of the Dose (NL-CM:11.1.4) and Administrator (NL-CM:11.1.6) concepts which are, based on their current definition, not suitable for future vaccinations. Therefore they have only been mapped to the Immunization profile.&#xD;&#xA;&#xD;&#xA;Please note that in contrary to the related zib concepts, `.vaccineCode` and `.occurrence[x]` are mandatory by FHIR and need to be recorded before Immunization resources can be exchanged." />
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-event.xml:35]: > to align with the intention of FHIR

Is dat niet bij elke zib zo? Dat we profilen om de zib met FHIR te alignen.

> while all future vaccinations, including the advised vaccinations, 

Waarom dit specifiek noemen? In de zib definitie gaat het alleen over 'administered' vs 'planned' vaccinations, dus ik zou ook alleen die twee termen gebruiken indien nodig.

>  Both profiles contain the zib mappings, with the exception of the Dose (NL-CM:11.1.4) and Administrator (NL-CM:11.1.6) concepts which are, based on their current definition, not suitable for future vaccinations. Therefore they have only been mapped to the Immunization

Gek dat er in Vaccination-event ingegaan wordt op twee concepten die niet in -request staan.

Dat verwerkend kom ik uit op iets als:

> The zib Vaccination is mapped to this Immunization profile and a profile on ImmunizationRecommendation (http://nictiz.nl/fhir/StructureDefinition/zib-Vaccination-request). All administered vaccinations are covered using this Immunization profile, while all planned vaccinations are covered in the ImmunizationRecommendation profile. This means that zib concepts can exist in both profiles.

> Please note that in contrary to the related zib concepts, `.vaccineCode` and `.occurrence[x]` are mandatory in FHIR and need to be recorded before Immunization resources can be exchanged.

File: resources/zib/zib-Vaccination-request.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
...sing the Immunization profile, while all future vaccinations, including the advised vaccinations, are covered in this ImmunizationRecommendation profile. Both profiles contain the zib mappings, except for the Dose (NL-CM:11.1.4) and Administrator (NL-CM:11.1.6) concepts which are, based on their current definition, not suitable for future vaccinations. Therefore they have only been mapped to the Immunization profile.&#xA;&#xA;Please note that the ImmunizationRecommendation profile has been aligned as much as possible with the zib Vaccination by setting the cardinalities of `.recommendation` ,`.vaccineCode` and `.dateCriterion` to x..1. This results in the removal of the ability to exchange a vaccination schema. Moreover, ImmunizationRecommendation has more mandatory elements that are not represented in the zib, such as `.date` and `.recomendation.forecastStatus`. Implementers will need to take these requirements into account when implementing the future vaccinations.&#xD;&#xA;&#xD;&#xA;The zib centre has acknowledged that the Vaccination zib is currently ill-defined, especially concerning future vaccinations. Therefore the zib will be completely redesigned for a future release." />
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-request.xml:35]: De feedback op -event hier toegepast:

> The zib Vaccination is mapped to this ImmunizationRecommendation profile and a profile on Immunization (http://nictiz.nl/fhir/StructureDefinition/zib-Vaccination-event). All administered vaccinations are covered using this Immunization profile, while all planned vaccinations are covered in the ImmunizationRecommendation profile. This means that zib concepts can exist in both profiles. The Dose (NL-CM:11.1.4) and Administrator (NL-CM:11.1.6) concepts are not suitable for planned vaccinations based on their current definitions and are therefore not mapped in this ImmunizationRecommendation profile.

> Please note that the ImmunizationRecommendation profile has been aligned as much as possible with the zib Vaccination by setting the maximum cardinalities of `.recommendation` ,`.vaccineCode` and `.dateCriterion` to 1. This results in the removal of the ability to exchange a vaccination schema. Moreover, ImmunizationRecommendation has mandatory elements that are not represented in the zib, such as `.date` and `.recomendation.forecastStatus`. Implementers will need to take these requirements into account when implementing the future vaccinations.

(Kleine nuanceverbeteringen in 2e paragraaf, 3e paragraaf bewust weggelaten - is het nodig om in dit profiel in te gaan op eventuele potentiële mogelijke wijzigingen in de zib in de toekomst?

File: resources/zib/zib-Vaccination-event.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      <path value="Immunization.status" />
+      <comment value="This element has no functional counterpart in the zib and cannot be implicitly mapped to other concepts. Unless a more appropriate status is recorded, it is assumed that only _completed_ Immunization resources will be exchanged." />
+    </element>
+    <element id="Immunization.vaccineCode">
+      <path value="Immunization.vaccineCode" />
+      <short value="ProductCode" />
+      <definition value="The product code of the vaccine administered.&#xD;&#xA;There are several possible code systems for documenting the product code. If the vaccination data is registered based on an anamnesis, coding with the ATC code is preferred. In all cases it concerns those products that fall under ATC group J07 (vaccines)." />
+      <alias value="ProductCode" />
+      <binding>
+        <strength value="example" />
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-event.xml:53]: Klopt deze binding? De individuele lijsten zijn Required

File: resources/zib/zib-Vaccination-event.xml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        <valueSet value="http://decor.nictiz.nl/fhir/ValueSet/2.16.840.1.113883.2.4.3.11.60.121.11.12--20200901000000" />
+      </binding>
+      <mapping>
+        <identity value="zib-vaccination-v4.0-2020EN" />
+        <map value="NL-CM:11.1.2" />
+        <comment value="ProductCode" />
+      </mapping>
+    </element>
+    <element id="Immunization.occurrence[x]">
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-event.xml:62]: `.patient` is 1..1, dus misschien zib-Patient al als targetProfile toevoegen?

File: resources/zib/zib-Vaccination-event.xml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      <path value="Immunization.doseQuantity" />
+      <short value="Dose" />
+      <definition value="The amount of product administered shown in milliliters. In most cases, the entire product is administered; in some cases, a described part of the product is administered." />
+      <alias value="Dosis" />
+      <mapping>
+        <identity value="zib-vaccination-v4.0-2020EN" />
+        <map value="NL-CM:11.1.4" />
+        <comment value="Dose" />
+      </mapping>
+    </element>
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-event.xml:97]: Als de definition zegt dat het over mL gaat, kunnen we dan niet de unit (en system, etc.) slimmer maken?

File: resources/zib/zib-Vaccination-event.xml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    <element id="Immunization.performer">
+      <path value="Immunization.performer" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="$this" />
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-event.xml:103]: Klopt deze discriminator als je slicet op basis van `.function`?

File: resources/zib/zib-Vaccination-request.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      <alias value="Vaccinatie" />
+      <mapping>
+        <identity value="zib-vaccination-v4.0-2020EN" />
+        <map value="NL-CM:11.1.1" />
+        <comment value="Vaccination" />
+      </mapping>
+    </element>
+    <element id="ImmunizationRecommendation.recommendation">
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-request.xml:43]: `.patient` op zib-Patient?

File: resources/zib/zib-Vaccination-request.xml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    </element>
+    <element id="ImmunizationRecommendation.recommendation.dateCriterion.value">
+      <path value="ImmunizationRecommendation.recommendation.dateCriterion.value" />
+      <short value="VaccinationDate" />
+      <definition value="Date (and if possible time) that the vaccine was or will be administered. In the case of a planned re-vaccination, a vague date (month, year) is allowed (Clinical reminder). If a planned vaccination cannot be administered before a certain date, the date stated (vague or not) must indicate this." />
+      <comment value="Please note that `.dateCriterion.code` is made mandatory by FHIR to provide a classification of the date. For example, earliest date to give, latest date to give, etc. This seems to be functionally covered by the zib as well, however, within the VaccinationDate concept as can be read from the definition: &quot;_If a planned vaccination cannot be administered before a certain date, the date stated (vague or not) must indicate this._&quot;. In FHIR it is not possible to include this kind of information inside a DateTime datatype, but rather it is explicitly captured in the `.dateCriterion.code` element." />
```
Comment (jduwel) [inline resources/zib/zib-Vaccination-request.xml:75]: > by FHIR

_in_ FHIR

> For example, earliest date to give, latest date to give, etc. 

Overbodig?

>This seems to be functionally covered by the zib as well, however, within the VaccinationDate concept as can be read from the definition: &quot;_If a planned vaccination cannot be administered before a certain date, the date stated (vague or not) must indicate this._&quot;. 

Beetje vaag taalgebruik

Alles verwerkt:

> Please note that `.dateCriterion.code` is made mandatory in FHIR to provide a classification of the date. This is functionally covered by the zib within the VaccinationDate concept as vague dates are allowed. This information cannot be included inside the DateTime datatype, but is explicitly captured in the `.dateCriterion.code` element.

File: resources/zib/zib-Vaccination-event.xml (hunk 1 above)
Comment (ArdonToonstra) [reply resources/zib/zib-Vaccination-event.xml:35]: Ok! Eens, is bondiger en duidelijker. Ik nam hier als voorbeeld Procedure, daar kan denk ik ook nog wat geschaafd worden.

File: resources/zib/zib-Vaccination-event.xml (hunk 4 above)
Comment (ArdonToonstra) [reply resources/zib/zib-Vaccination-event.xml:62]: Done!

File: resources/zib/zib-Vaccination-event.xml (hunk 5 above)
Comment (ArdonToonstra) [reply resources/zib/zib-Vaccination-event.xml:97]: Done! Goeie.

File: resources/zib/zib-Vaccination-event.xml [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+  <kind value="resource" />
+  <abstract value="true" />
+  <type value="Immunization" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Immunization" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Immunization">
+      <path value="Immunization" />
+      <short value="Vaccination" />
+      <comment value="The zib Vaccination is mapped to this Immunization profile and a profile on ImmunizationRecommendation (http://nictiz.nl/fhir/StructureDefinition/zib-Vaccination-request). All administered vaccinations are covered using this Immunization profile, while all planned vaccinations are covered in the ImmunizationRecommendation profile. &#xD;&#xA;&#xD;&#xA;Please note that in contrary to the related zib concepts, `.vaccineCode` and `.occurrence[x]` are mandatory in FHIR and need to be recorded before Immunization resources can be exchanged." />
```
Comment (LuudSlagter) [inline resources/zib/zib-Vaccination-event.xml:35]: Replace 'in contrary to' by 'contrary to'.

File: resources/zib/zib-Vaccination-request.xml [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+        <identity value="zib-vaccination-v4.0-2020EN" />
+        <map value="NL-CM:11.1.3" />
+        <comment value="VaccinationDate (implicit, main mapping is on `.dateCriterion.value`)" />
+      </mapping>
+    </element>
+    <element id="ImmunizationRecommendation.recommendation.dateCriterion.value">
+      <path value="ImmunizationRecommendation.recommendation.dateCriterion.value" />
+      <short value="VaccinationDate" />
+      <definition value="Date (and if possible time) that the vaccine was or will be administered. In the case of a planned re-vaccination, a vague date (month, year) is allowed (Clinical reminder). If a planned vaccination cannot be administered before a certain date, the date stated (vague or not) must indicate this." />
+      <comment value="Please note that `.dateCriterion.code` is made mandatory in FHIR to provide a classification of the date. This is functionally covered by the zib within the VaccinationDate concept as vague dates are allowed. This information cannot be included inside the DateTime datatype, but is explicitly captured `.dateCriterion.code`." />
```
Comment (LuudSlagter) [inline resources/zib/zib-Vaccination-request.xml:87]: Replace 'captured' by 'captured in'.

**Comment by pieter-edelman-nictiz** [comment]
Handmatig gemerged, ik sluit deze PR

**Pull request closed without merging**